planner/core/preprocess.go

```diff
@@ -577,8 +577,12 @@ func (p *preprocessor) checkAdminCheckTableGrammar(stmt *ast.AdminStmt) {
 			return
 		}
 		tempTableType := tableInfo.Meta().TempTableType
-		if stmt.Tp == ast.AdminCheckTable && tempTableType != model.TempTableNone {
-			p.err = infoschema.ErrAdminCheckTable
+		if (stmt.Tp == ast.AdminCheckTable || stmt.Tp == ast.AdminChecksumTable) && tempTableType != model.TempTableNone {
+			if stmt.Tp == ast.AdminChecksumTable {
+				p.err = ErrOptOnTemporaryTable.GenWithStackByArgs("admin checksum table")
+			} else {
+				p.err = infoschema.ErrAdminCheckTable
```

Comment:
```suggestion
				p.err = ErrOptOnTemporaryTable.GenWithStackByArgs("admin table")
```

Reply:
`ErrAdminCheckTable` is another kind of error, it means the `admin check table` fails because of the table data corruption.
Here we should report something like "unsupported operation", although the operation fails and returns error, the reason is different.